### PR TITLE
add-support-for-riscv

### PR DESCRIPTION
**Pull request by TONG369**
The PR submitted  is applicable to support RISCV64

### CODE REVIEW

**Review by penpornk — COMMENTED** [review]
Thank you for the PR! :) Could you please give us more context? 
1. What are you aiming to do with this PR? What does this PR, on its own, enable?
2. If this PR is the first of a series of PRs, what is the overall goal/plan?
    * Asking just to decide the scope of the work, and whether it will benefit from having a design review.
4. Do you plan to eventually use [oneDNN custom ops](https://github.com/tensorflow/community/blob/master/rfcs/20210930-enable-onednn-ops.md#design-proposal) or do you only use oneDNN through [Eigen contraction kernel](https://github.com/tensorflow/community/blob/master/rfcs/20210930-enable-onednn-ops.md#adding-support-to-eigen)? 
5. Both patches aren't currently in either repositories ([oneDNN](https://github.com/oneapi-src/oneDNN) and [rules_go](https://github.com/bazelbuild/rules_go)). Do you plan on upstreaming the changes? 
    * Asking because we prefer hosting patches only if they are specific to TF (e.g., the changes aren't applicable to the upstream repositories) or if they are only temporary (e.g., will be upstreamed soon, or already upstreamed but pending version update). 
    * RISC-V support seems beneficial to the upstream repositories as well.

File: third_party/0001-hardcode-go-SDKS.patch [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@@ -43,6 +43,14 @@ def go_host_sdk(name, register_toolchains = True, **kwargs):
+     if register_toolchains:
+         _register_toolchains(name)
+ 
++GO_SDKS = {
++ "linux_amd64": ("go1.18.4.linux-amd64.tar.gz", "c9b099b68d93f5c5c8a8844a89f8db07eaa58270e3a1e01804f17f4cf8df02f5"),
++ "linux_arm64": ("go1.18.4.linux-arm64.tar.gz", "35014d92b50d97da41dade965df7ebeb9a715da600206aa59ce1b2d05527421f"),
++ "linux_armv6l": ("go1.18.4.linux-armv6l.tar.gz", "7dfeab572e49638b0f3d9901457f0622c27b73301c2b99db9f5e9568ff40460c"),
++ "linux_riscv64":("go1.21rc2.linux-riscv64.tar.gz","33ea0c09d1f10f034d9e4fbef4eaceb785f3433ba631eaacb01a0af40773b673")
++}
```
[the remark this answers is not in the document]
Comment (cantonios) [reply third_party/0001-hardcode-go-SDKS.patch:23]: No, but this should probably be reviewed by someone on the bazel team.

[the remark this answers is not in the document]
Comment (TONG369) [reply third_party/0001-hardcode-go-SDKS.patch:23]: 1.  This PR can be used on target RISC-V
2. The overall goal is to support RISC-V with TensorFlow，I will update the code in a few days
3. use oneDNN custom ops
4. will be upstreamed soon

File: third_party/mkl_dnn/0001-new-error-impossible-constraint-in-asm-and-SYS_arch.patch [hunk 2]
```diff
@@ -0,0 +1,64 @@
+From 104102f2bedaca0f5832628365082bfedfb207c9 Mon Sep 17 00:00:00 2001
+From: xiaoyu_tong <tongxiaoyu@iscas.ac.cn>
+Date: Wed, 12 Jul 2023 06:05:56 +0000
+Subject: [PATCH] new-error-impossible-constraint-in-asm-and-SYS_arch
```
[the remark this answers is not in the document]
Comment (gbaned) [reply third_party/mkl_dnn/0001-new-error-impossible-constraint-in-asm-and-SYS_arch.patch:4]: Hi @TensorFlow-MKL Any update on this PR? Please. Thank you!

[the remark this answers is not in the document]
Comment (gbaned) [reply third_party/mkl_dnn/0001-new-error-impossible-constraint-in-asm-and-SYS_arch.patch:4]: Hi @TensorFlow-MKL Any update on this PR? Please. Thank you!

[the remark this answers is not in the document]
Comment (gbaned) [reply third_party/mkl_dnn/0001-new-error-impossible-constraint-in-asm-and-SYS_arch.patch:4]: Hi @TensorFlow-MKL Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @TONG369 Can you please resolve conflicts? Thank you!

**Comment by gbaned** [comment]
Hi @TONG369 Can you please resolve conflicts? Thank you!

**Comment by gbaned** [comment]
Hi @TONG369 Can you please resolve conflicts? Thank you!

**Pull request closed without merging**